fedimint-client/src/lib.rs

```diff
@@ -1098,14 +1118,18 @@ impl ClientInner {
                     missing_amount,
                 )
                 .await?;
-            partial_transaction.inputs.push(input);
+            partial_transaction.inputs.extend(inputs);
         }
 
-        let mut change_idx: Option<u64> = None;
+        let mut change_range = Range {
```

Comment:
**suggestion:** This being mutable, just to update it later is confusing to the reader. Probably better to just save the `range_start = ...` here, and construct the range at the end.

Reply:
I tried to avoid mutable state here but in this case it actually seemed like the the cleanest way to do it imo.